bson/json_util.py

```diff
@@ -824,22 +838,74 @@ def _parse_canonical_maxkey(doc: Any) -> MaxKey:
     return MaxKey()
 
 
+def _encode_bson(obj: Any, json_options: JSONOptions) -> Any:
+    type_marker = obj._type_marker
+    try:
+        return _encoders.get(type_marker)(obj, json_options)  # type: ignore
```

Comment:
Why the `# type: ignore`?

Reply:
The typechecker sees that the `get` call can return `None`, so it complains even with the try block around it.

Reply:
Oh but that's a bug that we need to guard against. It should be similar to our BSON logic here: https://github.com/mongodb/mongo-python-driver/blob/6537415da70c2a15cad52ebe7669673947ace65a/bson/__init__.py#L934-L935

Reply:
Whoops, it's meant to look like this: 

```
_encoders[type_marker](obj, json_options)  # type: ignore[no-untyped-call]
```

That way if `type_marker` isn't a valid type we support, it will raise a `KeyError` and be caught in the try block.